src/components/Plants/index.tsx

```diff
@@ -29,6 +43,52 @@ export default function PlantsDashboard(props: PlantsDashboardProps): JSX.Elemen
     populatePlantingSites();
   }, [organization]);
 
+  useEffect(() => {
+    const populateResults = async () => {
+      if (selectedPlantingSite) {
+        const serverResponse: PlantingSitesPlotsSearch[] | null = (await search({
+          prefix: 'plantingSites.plantingZones.plots',
+          fields: [
+            'id',
+            'fullName',
+            'populations.species_scientificName',
+            'populations.species_commonName',
+            'populations.totalPlants',
+          ],
+          search: {
+            operation: 'field',
+            field: 'plantingSite_id',
+            values: [selectedPlantingSite.id],
+          },
+          count: 0,
+        })) as unknown as PlantingSitesPlotsSearch[] | null;
+
+        if (serverResponse) {
+          let totalPlantsOfSite = 0;
+          // eslint-disable-next-line
+          const plantsPerSpecies: { [key: string]: number } = serverResponse.reduce((acc, plot) => {
```

Comment:
suggestion to break this up.
1) set the server response in state
2) have 3 useMemo functions
   a) const totalPlants = useMemo(() ..., [searchResults]);
   b) const totalPlantsBySpeciesByPlot = useMemo()...
   c) const totalPlantsBySpecies ....
this way we can have cleaner separation in code

Reply:
after looking at the other reviews, suggestion is to have some separation between index.tsx and what the other components do.
index.tsx
has a Map component (i can take care of) that accepts input on totalPlantsBySpeciesByPlots
has two chart components
1) first component accepts input totalPlantsBySpecies
2) second chart component accepts input totalPlantsBySpeciesByPlots - and internally it can handle the plot switching etc.
The total # of plants box looks good as-is.

Reply:
here's one way

```
useEffect(() => {
  search();
  setResults() or handleError();
}, [organization.id]);

const totalCount = useMemo(...), [searchResults, selectedPlantingSite.id])
const plantsBySpecies = useMemo(...)[searchResults, selectedPlantingSite.id]);
const plantsBySpeciesByPlot = useMemo(...)[searchResults, selectedPlantingSite.id]);

if (searching) {
  return <spinner>;
} 

if (!results) {
  return <emptystate>; //another jira
}


return ( 
  <>
    <Map data={plantsBySpeciesByPlot />
    <TotalCount data={totalCount}> // (having a Box is fine but separating out all the semantics into a separate component will be cleaner)
    <Chart-A data={plantsBySpecies} />
    <Chart-B data={plantsBySpeciesByPlot} />
  </>
)
```
that's just some pseudocode to help illustrate separation

Reply:
You can go one step further and have a separate component that handles everything for a selected planting site (which is how I would do it).

```
if (searching) {
  return <spinner>;
}

if (!results) {
  return <emptystate>; //another jira
}

return (
   <PlantingSiteDetails data={searchResults} />
)

const PlantingSiteDetails = (...) => {

const totalCount = useMemo(...), [data])
const plantsBySpecies = useMemo(...)[data]);
const plantsBySpeciesByPlot = useMemo(...)[data]);

return ( 
  <>
    <Map data={plantsBySpeciesByPlot />
    <TotalCount data={totalCount}> // (having a Box is fine but separating out all the semantics into a separate component will be cleaner)
    <Chart-A data={plantsBySpecies} />
    <Chart-B data={plantsBySpeciesByPlot} />
  </>
)
}
```